Crash upon .psbt import

electrum-dash crashes immediately upon 
Tools -> Load tx -> From file (pick foo.psbt)

Thanks

Tried to load the .psbt again in v. 4.0.9.0.
It still crashes. Now the Traceback is:

Traceback (most recent call last):

  File "electrum_dash\gui\qt\main_window.py", line 2968, in do_process_from_file

  File "electrum_dash\gui\qt\main_window.py", line 1202, in show_transaction

  File "electrum_dash\gui\qt\transaction_dialog.py", line 86, in
show_transaction

  File "electrum_dash\gui\qt\transaction_dialog.py", line 659, in __init__

  File "electrum_dash\gui\qt\transaction_dialog.py", line 395, in update

  File "electrum_dash\transaction.py", line 785, in estimated_size

  File "electrum_dash\transaction.py", line 819, in estimated_weight

  File "electrum_dash\transaction.py", line 809, in estimated_total_size

  File "electrum_dash\transaction.py", line 741, in serialize_to_network

  File "electrum_dash\transaction.py", line 741, in <genexpr>

  File "electrum_dash\transaction.py", line 738, in create_script_sig

  File "electrum_dash\transaction.py", line 678, in input_script

  File "electrum_dash\transaction.py", line 481, in multisig_script

AssertionError: m 0, n 1




On Fri, Jan 1, 2021 at 9:44 PM zebra-lucky <notifications@github.com> wrote:

> I wonder what would be the issue re-using such an address regardless of
> debug dump being publicly known? I don't see how that compromises DASH on
> this multi-sig address (beyond privacy, obviously). Am I missing something?
>
> Address info and txid info is private so there is need
> to improve code to strip this info from traces in future.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/akhavr/electrum-dash/issues/161#issuecomment-753380097>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASJTCCAF47BJL5CD3XR5NWDSXYX2FANCNFSM4VNV4UMA>
> .
>

--- a/electrum_dash/transaction.py
+++ b/electrum_dash/transaction.py
@@ -671,6 +671,12 @@ class Transaction:
         pubkeys, sig_list = self.get_siglist(txin, estimate_size=estimate_size)
         if _type in ('address', 'unknown') and estimate_size:
             _type = self.guess_txintype_from_address(txin.address)
+            if _type == 'p2sh':
+                # mimic upstream code for p2wpkh-p2sh type
+                # next two lines use code from upstream p2wpkh_nested_script
+                pkh = hash_160(bfh(pubkeys[0]))
+                redeem_script = construct_script([0, pkh])
+                return construct_script([redeem_script])
         if _type == 'p2pk':
             return construct_script([sig_list[0]])
         elif _type == 'p2sh':

